recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
```

Comment:
Looks like you are trying to get all the classes.  There is a cleaner way to do so using something called reflection (code looks at code, here to populate a list).

Reply:
Hi Prof Wu, thanks for reviewing my code!

This `__all__` is used to exclude `TwoChildrenEvent` from being exported. I think `TwoChildrenEvent` should be internal. An alternative to writing it this way to name the class `_TwoChildrenEvent` and remove the `__all__`. Python export everything except variables and classes that start with `_` by default. I also think it's clean to have an export statement at the start of the file so that people know what this file is about. Do you think I should remove this line?

Reply:
If TwoChildrenEvent is private (nobody outside the library should access it), then I think it is good to prefix with _.  It is fine to keep __all__ at the top, and I understand the rationale.  Mostly relies on user to keep __all__ up-to-date with the publicly accessible parts of library.  I think PEP standard is to not allow "from module import *", which is where __all__ is used, so maybe this is moot is people use the module correctly.  Sorry, don't know the standard...

Reply:
I agree with making `TwoChildrenEvent` private by using `_`. I don't mind keeping or removing `__all__`. I think it can serve the role of a comment, but is not strictly necessary..